### ISSUE TITLE
[ts, jsx] parser OOM crash on "0 ? (<b>0</b>) : (<b>0</b>)"

### ISSUE THREAD

**Issue by jeremyBanks** [facebookexperimental/rome#342]
👋 

## Description:

`@romejs/js-parser` with the `['ts', 'jsx']` syntaxes enabled crashes with a Node OOM error when you try to parse the document `0 ? (<b>0</b>) : (<b>0</b>)`. This does not happen with `['flow', 'jsx']` enabled instead.

## Steps To Reproduce

Tested Rome master at 964fde9a180b4d317386662d1c0d26f01ce84b84 (Sun Apr 26 17:55:48 2020 -0700) on Node v14.0.0.

1. ```bash
    echo '0 ? (<b>0</b>) : (<b>0</b>)' > example.tsx;
    rome parse ./example.tsx;
    ```
2. Observe a Node OOM error:
```
<--- Last few GCs --->

[19422:0x1045e7000]    25677 ms: Mark-sweep 2022.2 (2050.8) -> 2020.9 (2051.0) MB, 1731.6 / 0.0 ms  (average mu = 0.199, current mu = 0.063) allocation failure scavenge might not succeed
[19422:0x1045e7000]    27911 ms: Mark-sweep 2022.1 (2051.3) -> 2021.3 (2051.5) MB, 2231.4 / 0.0 ms  (average mu = 0.102, current mu = 0.001) allocation failure scavenge might not succeed

<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x1012303e5 node::Abort() (.cold.1) [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
 2: 0x10009d2c9 node::Abort() [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
 3: 0x10009d42f node::OnFatalError(char const*, char const*) [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
 4: 0x1001d6787 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
 5: 0x1001d6727 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
 6: 0x100369255 v8::internal::Heap::FatalProcessOutOfMemory(char const*) [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
 7: 0x10036aaea v8::internal::Heap::RecomputeLimits(v8::internal::GarbageCollector) [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
 8: 0x100366a45 v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
 9: 0x1003644ce v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
10: 0x10037180a v8::internal::Heap::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
11: 0x100371891 v8::internal::Heap::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
12: 0x10033dc77 v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationType, v8::internal::AllocationOrigin) [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
13: 0x1006a4720 v8::internal::Runtime_AllocateInOldGeneration(int, unsigned long*, v8::internal::Isolate*) [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
14: 0x1009eff59 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_NoBuiltinExit [/Users/jbanks/.nvm/versions/node/v14.0.0/bin/node]
error Command failed with signal "SIGABRT".
```

## Expected Results

The program would return a result instead of crashing.

**Comment by sebmck** [comment]
Looks like we're trying to parse it as an arrow function and it's hitting EOF and getting into an infinite loop parsing and creating nodes.

**Comment by sebmck** [comment]
Thanks for the report! Fixed by #343
